backend/mora/triggers/internal/amqp_trigger.py

```diff
@@ -0,0 +1,166 @@
+#
+# Copyright (c) Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+import logging
+import json
+import pika
+from mora import exceptions
+from mora import util
+from mora import mapping
+from mora.triggers import Trigger
+from mora.service.handlers import RequestType
+
+logger = logging.getLogger("amqp")
+amqp_connection = {}
+
+_SERVICES = ("employee", "org_unit")
+_OBJECT_TYPES = (
+    "address",
+    "association",
+    "employee",
+    "engagement",
+    "it",
+    "leave",
+    "manager",
+    "org_unit",
+    "related_unit",
+    "role",
+)
+_ACTIONS = ("create", "delete", "update")
+
+
+def publish_message(service, object_type, action, service_uuid, date):
+    """Send a message to the MO exchange.
+
+    For the full documentation, refer to "AMQP Messages" in the docs.
+    The source for that is in ``docs/amqp.rst``.
+
+    Message publishing is a secondary task to writting to lora. We
+    should not throw a HTTPError in the case where lora writting is
```

Comment:
Isn't this a change in semantics? I remember discussing this with @carlbordum when they wrote the AMQP, and I believe one of the requirements was that AMQP failures should trigger MO errors. Is that no longer the case?

Reply:
Jeg har, så vidt jeg selv kan gemmenskue :-) ikke ændret i Carls kode udover at binde den ind i en trigger, og det triggere er bedst til er at kaste exceptions. Faktisk er det noget jeg tror vi kommer til at bruge dem til - navnevalidering er noget af det, jetg har smoketested med:

```
@Trigger.on(mapping.ORG_UNIT, RequestType.CREATE, Trigger.Event.ON_BEFORE)
def ou_before_create(trigger_dict):

    if not namecheck(trigger_dict):
        ErrorCodes.E_INVALID_INPUT(
            'department prefix not allowed - must be in {}'.format(
                allowed_department_prefixes
            )
        )
```